### PR TITLE
ICS-3: update `Version` field

### LINKED ISSUE #809: Connection `version` still defined with `string`: inconsistent with ibc-go
The connection version in the standard is inconsistent with what's being used in ibc-go. Specifically, in the standard, a version is an opaque `string`, whereas in ibc-go, it is as follows:

```proto
message Version {
  option (gogoproto.goproto_getters) = false;

  // unique version identifier
  string identifier = 1;
  // list of features compatible with the specified identifier
  repeated string features = 2;
}
```

My understanding is that the standard fell behind and should be updated to reflect what's in ibc-go.

Additionally, the values for currently standard identifier/features should be added to the standard:

+ identifier: `"1"` to specify IBC 1.0.0
+ features: `"ORDER_UNORDERED"` and `"ORDER_ORDERED"` to specify unordered and ordered channels, respectively.

The signature of helper functions `getCompatibleVersions` and `pickVersion` should be adjusted accordingly.

I'm happy to open a PR with the fix if we agree that the standard should indeed be fixed as described.

### PR DESCRIPTION
**Pull request by plafer**
Closes #809

### CODE REVIEW

**Comment by plafer** [comment]
The CI claims that https://blog.cosmos.network/eli5-what-is-ibc-a212f518715f is dead, but it works when I try it.

**Review by AdityaSripal — APPROVED** [review]
Thanks @plafer !

**Review by mpoke — APPROVED** [review]
Nice work @plafer.
